src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +446,82 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool allocation_at_init = !LOG_ISRESTARTED ();
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_total_mem_usage += size;
+
+    m_module[module_idx]->add_stat (size, comp_idx, subcomp_idx, allocation_at_init);
+
+    if (thread_p)
```

Comment:
About tracking the transaction memory usage
1. it would be better introduce a new function for this
    - for example
        ```
        int memory_monitor::add_tran_stat (THREAD_ENTRY *thread_p, unit64_t size);
        {
            // put the code in here
        }
        ```
2. consider this job(transaction memory usage) handling in the upper layer; like
    ```
    mmon_add_stat()
    {
        mmon_Gl->add_stat(); // responsibility for tracking the module memory usages
        mmon_Gl->add_tran_stat(); // responsibility for tracking the transaction memory usages
    }
    ```